core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -234,11 +242,31 @@ void processSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateIn
     }
   }
 
+  private void updateHealthCheckedState(SubchannelData subchannelData) {
+    if (subchannelData.state != READY) {
+      return;
+    }
+    SubchannelPicker picker = new Picker(PickResult.withNoResult());
```

Comment:
I'm wary of `nextConcludedState` being calculated separately from the Picker. How about:
```java
    if (subchannelData.getHealthState() == READY) {
      updateBalancingState(READY, new FixedResultPicker(PickResult.withSubchannel(subchannelData.subchannel)));
    } else if (subchannelData.getHealthState() == TRANSIENT_FAILURE) {
      updateBalancingState(TRANSIENT_FAILURE, new Picker(PickResult.withError(
          subchannelData.healthListener.healthStateInfo.getStatus())));
    } else {
      updateBalancingState(subchannelData.getHealthState(), new Picker(PickResult.withNoResult()));
    }
```